fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/MaterializedViewRewriter.java

```diff
@@ -785,7 +785,7 @@ protected Multimap<ScalarOperator, ColumnRefOperator> normalizeAndReverseProject
                 } else {
                     rewriteScalarOp = columnRewriter.rewriteViewToQueryWithQueryEc(rewritten);
                 }
-                reversedMap.put(rewriteScalarOp, entry.getKey());
+
                 // if rewriteScalarOp == rewritten, it means the rewritten can not be mapped from mv to query
                 // and ColumnRefOperator may conflict between mv and query(same id but not same name),
                 // so do not put it into the reversedMap
```

Comment:
why `isViewBased ` no need to check `rewriteScalarOp != rewritten`?

Reply:
you mean view against query? because there is no need to map from view to query.